Rubberduck.Parsing/Symbols/VariableDeclaration.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.Collections.Generic;
+using Antlr4.Runtime;
+using Rubberduck.Parsing.Annotations;
+using Rubberduck.Parsing.Grammar;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.VBEditor;
+
+namespace Rubberduck.Parsing.Symbols
+{
+    public sealed class VariableDeclaration : Declaration, ICanBeInterfaceMember
+    {
+        public VariableDeclaration(
+            QualifiedMemberName qualifiedName,
+            Declaration parentDeclaration,
+            Declaration parentScope,
+            string asTypeName,
+            string typeHint,
+            bool isSelfAssigned,
+            bool isWithEvents,
+            Accessibility accessibility,
+            ParserRuleContext context,
+            ParserRuleContext attributesPassContext,
+            Selection selection,
+            bool isArray,
+            VBAParser.AsTypeClauseContext asTypeContext,
+            IEnumerable<IAnnotation> annotations = null,
+            Attributes attributes = null)
+            : base(
+                qualifiedName,
+                parentDeclaration,
+                parentScope?.Scope,
+                asTypeName,
+                typeHint,
+                isSelfAssigned,
+                isWithEvents,
+                accessibility,
+                DeclarationType.Variable,
+                context,
+                attributesPassContext,
+                selection,
+                isArray,
+                asTypeContext,
+                true,
+                annotations,
+                attributes)
+        {
+            if ((accessibility == Accessibility.Public || accessibility == Accessibility.Implicit) 
```

Comment:
`Implcit` isn't an accessibility type - whether a declaration's accessibility is implicitly defined should be a property of the declaration, not forced as a psudo-type

Reply:
More tech debt. I believe that this is being driven by the resolver, but I agree that at some point we need to remove `Accessibility.Implicit` entirely.